sources/packages/backend/apps/api/src/services/student-appeal/student-appeal.service.ts

```diff
@@ -129,19 +129,17 @@ export class StudentAppealService extends RecordDataModelService<StudentAppeal>
   }
 
   /**
-   * Find any pending appeal for a student if exists.
-   * @param userId of student.
+   * Find any pending appeal for the application if exists.
+   * @param applicationId application id related to the appeal.
    * @returns exist status
    */
-  async hasExistingAppeal(userId: number): Promise<boolean> {
+  async hasExistingAppeal(applicationId: number): Promise<boolean> {
     const existingAppeal = await this.repo
       .createQueryBuilder("appeal")
       .select("1")
       .innerJoin("appeal.appealRequests", "appealRequests")
       .innerJoin("appeal.application", "application")
-      .innerJoin("application.student", "student")
-      .innerJoin("student.user", "user")
-      .where("user.id = :userId", { userId })
+      .andWhere("application.id = :applicationId", { applicationId })
```

Comment:
Is it not missing the `where` before the first `andWhere` is added?
Is it producing the expected SQL?

Reply:
That is a mistake. I have changed it to `where`.  Below is the generated query after the change:

<img width="1381" alt="image" src="https://github.com/bcgov/SIMS/assets/7859295/7fa70147-5035-4baf-a40d-b2be1c8f63ce">